### PR TITLE
Codegen: convert Arc<Box<T>> to Arc<T>

### LINKED ISSUE #234: Codegen with `Arc<dyn ..>` instead of `Arc<Box<..>>`
Hi, I am using ttrpc for some time. One thing confused me is that with ttrpc's `Codegen`, we would generate a `.._ttrpc.rs` where a `XXXMethod` would have a `service` member with type `Arc<Box<dyn ...>>` rather than `Arc<dyn ...>`.

I wonder if we could convert `Arc<Box<dyn ...>>` to `Arc<dyn ...>`?

The reason comes from some ttrpc server implementation.

We are using a proto with more-than-one Service, e.g.
```protobuf
service A {
    rpc funcA(ARequest) returns (AResponse) {};
}

service B {
    rpc funcB(BResquest) returns (BResponse) {};
}
```

We want that a same object would serve the two APIs, thus we have
```rust
// Server impls A and B
let server = Server::new();
let server = Arc::new(Box::new(server as _));

let a = create_a_service(server.clone());

// compiler errors because we cannot convert Arc<Box<T>> to both Arc<Box<dyn A>> and Arc<Box<dyn B>>
let b = create_a_service(server);
```

So what we can do is only to create two `Server`s to create different services, which means double the memory consumption.

If the ttrpc CodeGen returns type `Arc<dyn T>`, we can facilitate by
```rust
let server = Arc::new(Server::new());

let a = create_a_service(server.clone() as _);
let b = create_b_service(server as _);

let mut server = TtrpcServer::new()
        .bind(&socket)
        .unwrap()
        .register_service(a)
        .register_service(b);
```
where the `as _` would help the compiler convert `Arc<Server>` to `Arc<dyn A>` and `Arc<dyn B>`.

### PR DESCRIPTION
**Pull request by Xynnn007**
This commit changes the generated ttrpc server from Arc<Box<T>> to Arc<T>. This helps the type conversion and also avoids extra runtime cost caused by double pointer.

Fixes #234

### CODE REVIEW

**Review by wllenyj — APPROVED** [review]

**Comment by Xynnn007** [comment]
@wllenyj sry please wait a moment. Let me test locally if sync version works fine with this.

**Comment by Xynnn007** [comment]
@wllenyj Add the line of sync version `create_xxx_server`

**Comment by Xynnn007** [comment]
Hi @wllenyj @Tim-Zhang What should I do else to move this forward?

**Review by Tim-Zhang — APPROVED** [review]
It will simplify the code:
before:
```
    let h = Box::new(HealthService {}) as Box<dyn health_ttrpc::Health + Send + Sync>;
    let h = Arc::new(h);
    let hservice = health_ttrpc::create_health(h);
```

after: 
```
    let hservice = health_ttrpc::create_health(Arc::new(HealthService {}));
```

it also will cut down heap allocations, thanks @Xynnn007

File: example/Cargo.toml [hunk 1]
```diff
@@ -51,3 +51,7 @@ path = "./async-stream-client.rs"
 
 [build-dependencies]
 ttrpc-codegen = { path = "../ttrpc-codegen"}
+
+[patch.crates-io]
+ttrpc-compiler = { path = "/home/tim/project/ttrpc-rust/compiler"}
```
Comment (Xynnn007) [inline example/Cargo.toml:56]: This is for debug, right?

Comment (Tim-Zhang) [reply example/Cargo.toml]: No, but I'm still working on it.

**Pull request merged**